app/services/youtube_service.rb

```diff
@@ -11,9 +11,24 @@ def playlist_info(id)
 
   def playlist_items_info(id)
     results_count = playlist_info(id)[:items][0][:contentDetails][:itemCount]
+    results_count = 50 if results_count > 50
     params = { part: part_details, playlistId: id, maxResults: results_count }
-    get_json("youtube/v3/playlistItems", params)
-    #if there is a next_page token, iterage through and add it to the array
+    items = get_json('youtube/v3/playlistItems', params)
+    next_page_playlist_items(items, params)
+  end
+
+  def next_page_playlist_items(items, params)
+    return items unless items[:nextPageToken]
+
+    next_page_token = items[:nextPageToken]
+    while next_page_token
+      next_page_items = get_json("youtube/v3/playlistItems?pageToken=#{next_page_token}", params)
+      next_page_items[:items].each do |item|
+        items[:items] << item
+      end
+      next_page_token = next_page_items[:nextPageToken]
+    end
+    items
   end
```

Comment:
if there is a nextPageToken, then run the API again. then iterate through to add the new videos to the existing array of videos